docs/dev/start-guide.md

```diff
@@ -1,37 +1,39 @@
 # Onboarding to Atlas Operator
-
-1. Install Go (1.16)
-2. Install dependencies
-```
-# on Mac
-brew install kubebuilder # or https://book.kubebuilder.io/quick-start.html
-brew install kustomize # or https://kubectl.docs.kubernetes.io/installation/kustomize/
-brew install act # or https://github.com/nektos/act#installation
-brew install yq # or https://pypi.org/project/yq/
-brew install shellcheck # or https://github.com/koalaman/shellcheck#installing
-brew install coreutils # or https://www.gnu.org/software/coreutils/
-brew install pre-commit # or https://pre-commit.com/index.html#install
-pre-commit install # from the root of the project
-```
-3. Install Kind ([instructions](https://kind.sigs.k8s.io/docs/user/quick-start/#installation))
+1. Install Nix [[Instructions](https://nixos.org/download/)]
+2. Run the Nix shell (by entering ```shell.nix``` in the bash to initialize an environment with pre-installed dependencies). For more details, refer to the shell.nix file.
+3. Install optional dependencies
+    ```
+    # on Mac
+    brew install coreutils # or https://www.gnu.org/software/coreutils/
+    brew install pre-commit # or https://pre-commit.com/index.html#install
+    pre-commit install # from the root of the project
+    ```
+4. Install Kind [[Instructions](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)]
 4. Clone the project to your workspace (note, that this doesn't need to be `GOPATH` as the project uses Go Modules)
 5. Copy the default Github Actions settings for local run: `cp .actrc.local.sample .actrc`
 6. Copy the default Github Actions environment for local run: `cp dotenv.sample .env`
 7. Update the .actrc - specify your Atlas connectivity data (orgId, keys)
 8. Build and deploy the Operator into the K8s cluster: `make deploy`
-9. Create an AtlasProject: `kubectl apply -f config/samples/atlas_v1_atlasproject.yaml` (note, that Atlas connection secrets are
- created during running `make deploy`)
+9. Create an AtlasProject: `kubectl apply -f config/samples/atlas_v1_atlasproject.yaml` (note, that Atlas connection secrets are created during running `make deploy`)
 10. Create an AtlasDeployment: `kubectl apply -f config/samples/atlas_v1_atlasdeployment.yaml`
 11. Create an AtlasDatabaseUser: `kubectl apply -f config/samples/atlas_v1_atlasdatabaseuser.yaml`
 
 Some more details about using `act` can be found in [HOWTO.md](../../.github/HOWTO.md)
-
+### IDE setup with Nix
+For VSCode
+1. Install the extension [[Nix Environment Selector](https://marketplace.visualstudio.com/items?itemName=arrterian.nix-env-selector)] 
+2. Restart the IDE and press (Control + Shift + P) and search for Nix-Env: Select Environment command.
+3. Choose the Nix environment from the repo (shell.nix) and wait for it to build.
+4. Start using VSCode with everything installed.
 # How-To
 ## Run integration tests
+
+**IMPORTANT: Please ensure you are in a Nix environment when running any make targets**
+
```

Comment:
I assume this also needs to be updated to use Devbox?

Reply:
yes. thanks